plugins/couchbase/couchbaseembed.cpp

```diff
@@ -899,68 +899,145 @@ namespace couchbaseembed
 
     void CouchbaseRowBuilder::processBeginDataset(const RtlFieldInfo * field)
     {
-        /*
-         *
-         *childRec := RECORD real x; real y; END;
-         *parentRec := RECORD
-         * childRec child1,                        <-- flatens out the childrec, this function would receive a field of name x
-         * dataset(childRec) child2;               <-- keeps nested structure, this funciton would receive a field of name child2
-         *END;
-        */
+        const char * xpath = xpathOrName(field);
 
-        if (getNumFields(field->type->queryChildType()) > 0)
-            m_oNestedField.set(m_oResultRow->queryBranch(field->name->queryStr()));
+        if (xpath && *xpath)
+        {
+            PathTracker     newPathNode(xpath, true);
+            StringBuffer    newXPath;
+
+            constructNewXPath(newXPath, xpath);
```

Comment:
Unfortunately, yes. XPATHs can be relative (which are more common) or absolute, or at least "full" in the sense that they must start processing at the root of the property tree. This is a topic that may bear revisiting at some point in the future, though.